src/Utilities/Compiler/Extensions/INamedTypeSymbolExtensions.cs

```diff
@@ -139,8 +139,10 @@ public static bool IsStaticHolderType(this INamedTypeSymbol symbol)
                 return false;
             }
 
-            // Sealed objects are presumed to be non-static holder types
-            if (symbol.IsSealed)
+            // Sealed objects are presumed to be non-static holder types for C#.
+            // In VB.NET the type cannot be static and guidelines favor having a sealed (NotInheritable) type
+            //  to act as static holder type.
+            if (symbol.IsSealed && symbol.Language == LanguageNames.CSharp)
```

Comment:
Do you need to do changes to both these extension method and the analyzer? Can we not just do one of them?

Reply:
I need both of them because the helper is a little misleading, it's actually `IsStaticHolderLikeType` (or something like this). The helper checks that the member looks like a static holder but disregard the fact it is static to allow for the rule to report.

So in the end, the change here is just to prevent judging the type as not static-holder like when it's sealed. And the change in `StaticHolderTypes.cs` is made to ensure that if the class is sealed and it's vbnet we don't suggest to mark the class as static.